src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -243,7 +243,7 @@ void CalorimeterHitReco::process(
         std::vector<double> cdim;
         // get segmentation dimensions
         auto segmentation_type = m_converter->findReadout(local).segmentation().type();
-        if (segmentation_type == "CartesianGridXY" || segmentation_type == "HexGridXY") {
+        if (segmentation_type == "CartesianGridXY" || segmentation_type == "HexGridXY" || segmentation_type=="MultiSegmentation") {
```

Comment:
There's a reason for this `if`: It's here to make sure that we actually have a segmentation of a planar geometry with 2-dimensional cells where the X and Y coordinates give the size of the cell. That's true for CartesianGridXY and for HexGridXY (maybe). It's why we don't have CartesianGridXZ here, or any of the other non-XY segmentations.

This is simply not the right place to handle MultiSegmentation (though it may be for a MultiSegmentation iff the actual segmentation of the specific cellID is CartesianGridXY or HexGridXY).

Reply:
hmm... maybe we could check recursively if the underlying subsegmentation is either CartesianGridXY or HexGridXY?  Otherwise algorithms that require the cell size (like Hexplit) won't work for multisegmentation.

Reply:
I just checked in this recursive version... @wdconinc what do you think?